scripts/pre-commit.sh

```
@@ -0,0 +1,343 @@
+    read -rp "> " breaking_changes_text
+    breaking_changes_text="BREAKING CHANGE: $breaking_changes_text"
+  fi
+  
+  # Get issues closed (optional)
+  echo -e "\n${BOLD}${BLUE}Issues closed (optional, comma separated, press enter to skip):${NC}"
+  read -rp "> " issues_closed
+  
+  issues_text=""
+  if [ -n "$issues_closed" ]; then
+    # Convert comma-separated issues to format: "Closes #123, Closes #456"
+    issues_array=($(echo "$issues_closed" | tr ',' ' '))
```

Shellcheck (0.10.0)

[warning] Prefer mapfile or read -a to split command output (or quote to avoid splitting).

(SC2207)